custom_components/better_thermostat/climate.py

```diff
@@ -257,36 +257,46 @@ def _async_startup(*_):
 			self.hass.bus.async_listen_once(EVENT_HOMEASSISTANT_START, _async_startup)
 		
 		# Check If we have an old state
-		old_state = await self.async_get_last_state()
-		if old_state is not None:
+		if (old_state := await self.async_get_last_state()) is not None:
 			# If we have no initial temperature, restore
 			if self._target_temp is None:
 				# If we have a previously saved temperature
 				if old_state.attributes.get(ATTR_TEMPERATURE) is None:
-					self._target_temp = self.min_temp
+					self._target_temp = self._TRV_min_temp
 					_LOGGER.debug(
 						"better_thermostat %s: Undefined target temperature, falling back to %s", self.name, self._target_temp
 					)
 				else:
-					_old_target_temperature = float(old_state.attributes.get([ATTR_TEMPERATURE]))
+					_old_target_temperature = float(old_state.attributes.get(ATTR_TEMPERATURE))
+					self._target_temp = _old_target_temperature
+					# if the min_temp is setted by the config, use it
+					if self._min_temp is not None:
+						self._TRV_min_temp = self._min_temp
```

Comment:
that's not correct. We need to import the TRV minimum/maximum for mode 2. Otherwise we cannot set the TRV past our config setpoints for calibration.

Reply:
your right i missed it. sadly i want to review the min max PR because its more complicated than we think

give me some time to think about this.